src/libcore/char.rs

```diff
@@ -394,6 +429,25 @@ impl CharExt for char {
         EscapeDefault { state: init_state }
     }
 
+    #[stable(feature = "rust1", since = "1.0.0")]
+    fn escape_control(self) -> EscapeDefault {
+        let init_state = match self {
+            // Backslash-escape special control character.
+            '\t' => EscapeDefaultState::Backslash('t'),
+            '\r' => EscapeDefaultState::Backslash('r'),
+            '\n' => EscapeDefaultState::Backslash('n'),
+            '\\' => EscapeDefaultState::Backslash('\\'),
+            '\'' => EscapeDefaultState::Backslash('\''),
+            '"'  => EscapeDefaultState::Backslash('"'),
+            // Unicode-escape other C0 or C1 control character.
+            '\x00' ... '\x1f' | '\u{80}' ... '\u{9f}' => 
+                EscapeDefaultState::Unicode(self.escape_unicode()),
+            // Don't escape everything else.
```

Comment:
Thanks. In the linked issue I proposed the idea of just escaping C0 and C1 codes. Two people in the IRC agreed...